source/joker_effects.c

```diff
@@ -294,44 +326,55 @@ static JokerEffect blackboard_joker_effect(Joker *joker, Card *scored_card) {
     return effect;
 }
 
-static JokerEffect blue_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect blue_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
 
     effect.chips = (get_deck_top() + 1) * 2;
 
     return effect;
 }
 
-static JokerEffect raised_fist_joker_effect(Joker *joker, Card *scored_card) 
-{
+static JokerEffect raised_fist_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
 
-    // Find the lowest rank card in hand
-    // Aces are always considered high value, even in an ace-low straight
-    u8 lowest_value = IMPOSSIBLY_HIGH_CARD_VALUE;
-    CardObject** hand = get_hand_array();
-    int hand_size = hand_get_size();
-    for (int i = 0; i < hand_size; i++ )
-    {
-        u8 value = card_get_value(hand[i]->card);
-        if (lowest_value > value)
-            lowest_value = value;
-    }
+    // Correcting Raised Fist logic and preparing it for Held Cards scoring
 
-    if (lowest_value != IMPOSSIBLY_HIGH_CARD_VALUE)
-        effect.mult = lowest_value * 2;
+    switch (scored_when) {
+        // Optimization: search for the lowest value card index only once
+        // Resets after hand has finished playing to prepare for the next one
+        case JOKER_CALLBACK_ON_HAND_SCORED_END:
+            joker->data = -1;
+            break;
+        case JOKER_CALLBACK_ON_HAND_PLAYED:
+            CardObject** hand = get_hand_array();
+            int hand_size = hand_get_size();
+            for (int i = 0; i < hand_size; i++) {
+                u8 value = card_get_value(hand[i]->card);
+                // >= instead of > to match Balatro's logic of selecting the
+                // rightmost card among cards of same value
+                if (joker->data == -1 || joker->data >= value) {
+                    joker->data = i;
+                }
+            }
+            break;
+        case JOKER_CALLBACK_ON_CARD_HELD:
+            if (joker->data == get_scored_card_index()) {
+                CardObject** hand = get_hand_array();
+                effect.mult = 2 * card_get_value(hand[joker->data]->card);
+            }
+            break;
+            
+    }
 
     return effect;
-} 
+}
 
-static JokerEffect reserved_parking_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect reserved_parking_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_ON_CARD_HELD_ONLY(scored_card, scored_when, effect)
```

Comment:
```suggestion
    SCORE_ON_EVENT_ONLY(JOKER_CALLBACK_INDEPENDANT, scored_when, effect)
```

Notice that I changed back to the independent stage since JOKER_CALLBACK_ON_CARD_HELD is not yet implemented so actually **this change breaks Reserved Parking**.

Batch - Consolidate joker event restriction macros
Batch - Remove JOKER_CALLBACK_ON_CARD_HELD